ignite/contrib/handlers/trains_logger.py

```diff
@@ -659,32 +661,84 @@ def _setup_check_trains(self, logger, output_uri):
         if output_uri:
             self._task.output_uri = output_uri
 
+    class _CallbacksContext:
+        def __init__(
+            self, slots: List, checkpoint_key: str, filename: str, basename: str, metadata: Optional[Mapping] = None
+        ):
+            self._slots = slots
+            self._checkpoint_key = str(checkpoint_key)
+            self._filename = filename
+            self._basename = basename
+            self._metadata = metadata
+
+        def pre_callback(self, action: str, model_info: Any):
```

Comment:
@jkhenning what kind of object practically it is `model_info` ?
and as discussed with @bmartinn can we use an enum instead of `str` for checking `action == "save"` ?

Reply:
@vfdev-5 `model_info`'s type is an internal type (`WeightsFileHandler.ModelInfo`).
As for the enum - we already added that to Trains after we released the RC - I'll update it soon.